apps/guardian-ui/src/setup/FederationSetup.tsx

```diff
@@ -47,9 +47,17 @@ export const FederationSetup: React.FC = () => {
     dispatch,
     api,
   } = useSetupContext();
-  const [needsTosAgreement, setNeedsTosAgreement] = useState(!!getEnv().TOS);
+  const [needsTosAgreement, setNeedsTosAgreement] = useState(true);
   const [confirmRestart, setConfirmRestart] = useState(false);
 
+  useEffect(() => {
+    async function getTos() {
+      const tosPresent = !!(await getEnv()).tos;
+      setNeedsTosAgreement(tosPresent);
+    }
+    getTos();
+  }, []);
```

Comment:
It's a shame to see a bunch of these scattered throughout the app because each one makes a separate request, and the initial rendering state of these components can be wrong. I think it would make more sense to fetch the config once, show a loader while it's fetching, and then have a React context provide it to all components from a single spot.